packages/react-query/src/types.ts

```diff
@@ -155,7 +156,7 @@ export type UseQueryResult<
 export type UseSuspenseQueryResult<
   TData = unknown,
   TError = DefaultError,
-> = OmitKeyof<
+> = DistributiveOmit<
```

Comment:
@manudeli I think your `OmitKeyof` helper doesn’t work over union types (just like the build-in `Omit` also doesn’t work with it).

Because `DefinedQueryObserverResult` is a union of:

```
| QueryObserverRefetchErrorResult<TData, TError>
| QueryObserverSuccessResult<TData, TError>
```

it “doesn’t work” in a sense that type narrowing to `error` doesn’t happen anymore because it’s not distributive.

I’ve added the standard `DistributiveOmit` helper and things work fine, but if you can improve your helper type to not fail the added test case, we can switch back.